azure-pipelines.yml

```diff
@@ -0,0 +1,235 @@
+  # See https://docs.opensource.microsoft.com/tools/nuget_security_analysis.html for more info.
+  - name: NugetSecurityAnalysisWarningLevel
+    value: none
+  
+  # PTVS variable group, for secrets
+  # This group contains an AzDO Personal Access Token (ArtifactServices.Symbol.PAT) that expires on 4/15/2021 which is used to publish symbols to symweb.
+  # If Visual Studio insertion PRs start failing the Symbol Check step, this token is most likely expired.
+  # See https://www.1eswiki.com/wiki/Using_Azure_DevOps_Symbols for more information about how to create a new PAT and how to update the variable group.
+  - group: PTVS
+
+# TODO:
+# For PR builds, don't real sign. Test sign instead.
+
+steps:
+
+# check out code clean from source control
+- checkout: self
+  clean: true
+
+# skip some steps if this is a PR build
+- ${{ if notin(variables['Build.Reason'], 'PullRequest') }}:
+
+  # add build tag
+  - powershell: 'Write-Host "##vso[build.addbuildtag]$env:VSTarget"'
+    displayName: 'Add vstarget build tag'
+
+  # install plugins needed for swixproj/vsmanproj, signing, and localization
+  - task: MicroBuildSwixPlugin@3
+    displayName: 'Install swix plugin'
+    inputs:
+      dropName: 'Products/$(System.TeamProject)/$(Build.Repository.Name)/$(Build.SourceBranchName)/$(Build.BuildNumber)'
+      dropServiceUri: 'https://vsdrop.corp.microsoft.com/file/v1'
+      feedSource: 'https://mseng.pkgs.visualstudio.com/DefaultCollection/_packaging/MicroBuildToolset/nuget/v3/index.json'
+
+  - task: MicroBuildSigningPlugin@3
+    displayName: 'Install signing plugin'
+    inputs:
+      signType: 'Test' # TODO: Change this to $(SignType) once Real signing is approved for this AzDO project ID
+      feedSource: 'https://devdiv.pkgs.visualstudio.com/DefaultCollection/_packaging/MicroBuildToolset/nuget/v3/index.json'
+
+  - task: MicroBuildLocalizationPlugin@3
+    displayName: 'Install localization plugin'
+    inputs:
+      type: $(LocType)
+      languages: 'VS'
+      feedSource: 'https://mseng.pkgs.visualstudio.com/DefaultCollection/_packaging/MicroBuildToolset/nuget/v3/index.json'
+
+# Run a powershell script that does a package restore, some symlinking, and installs debugpy
+- task: PowerShell@1
+  displayName: 'Restore packages'
+  inputs:
+    scriptName: Build/PreBuild.ps1
+    arguments: '-vstarget $(VSTarget)'
+
+# Clean the old swix outputs
+- task: DeleteFiles@1
+  displayName: 'Clean old swix outputs'
+  inputs:
+    SourceFolder: '$(Build.BinariesDirectory)\raw\setup\swix'
+    Contents: '**'
+  continueOnError: true
+
+# Build the product
+- task: MSBuild@1
+  displayName: 'Build product'
+  inputs:
+    solution: Python/Product/dirs.proj
+    msbuildVersion: 16.0
+    platform: 'AnyCPU'
+    configuration: '$(BuildConfiguration)'
+
+# Build the installer
+- task: MSBuild@1
+  displayName: 'Build installer'
+  inputs:
+    solution: Python/Setup/dirs.proj
+    msbuildVersion: 16.0
```

Comment:
here was the second. but ok

Reply:
ah you're right, it's referenced twice. My mistake. Sure I can make it a variable then 😄